UM/Settings/Interfaces.py

```diff
@@ -35,6 +35,10 @@ def pushContainer(self, container):
     def popContainer(self):
         return self._stack_of_containers.pop()
 
+    @property
+    def context(self):
+        return self._context
```

Comment:
I'd suggest just making the field public rather than this construction.

Once you've done that it appears that the PropertyEvaluationContext is just a namedtuple for the context dictionary and the container stack.